pkg/api/v1alpha1/etcdencryption_types.go

```diff
@@ -0,0 +1,21 @@
+package v1alpha1
+
+// EtcdEncryption defines the configuration for ETCD encryption.
+type EtcdEncryption struct {
+	Providers []EtcdEncryptionProvider `json:"providers"`
+	// Resources defines a list of objects and custom resources definitions that should be encrypted.
+	Resources []string `json:"resources"`
+}
+
+// EtcdEncryptionProvider defines the configuration for ETCD encryption providers.
+// Currently only KMS provider is supported.
+type EtcdEncryptionProvider struct {
+	// KMS defines the configuration for KMS Encryption provider.
+	KMS KMS `json:"kms"`
```

Comment:
this probably need to be a pointer if we ever going to support different providers
So one can be filled and the other ones can be nill

Reply:
We don't have plans to support another encryption provider besides KMS. But I can it a pointer, shouldn't make too much difference.